### PR TITLE
APPEALS-52892: For all legacy appeal states/notifications check status in VACOLS

### PR DESCRIPTION
**Pull request by noahhansen-gov**
Resolves [For all legacy appeal states/notifications check status in VACOLS](https://jira.devops.va.gov/browse/APPEALS-52892)

# Description
An additional check is made within AppellantNotification#handle_errors
This check ensures that an appeal is active before sending a notification
Both Appeal and LegacyAppeal have access to the "active?" method that returns a boolean
If an appeal is active, the notification continues as normal
if an appeal is NOT active, we abort any further action with that notification

## Acceptance Criteria
- [x] Code compiles correctly

## Testing Plan
spec tests

# Best practices

### Test Coverage
Did you include any test coverage for your code? Check below:
- [x] RSpec
- [ ] Jest
- [ ] Other

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [x] No new code climate issues added

### CODE REVIEW

File: spec/models/appellant_notification_spec.rb [hunk 1]
```diff
@@ -936,6 +946,7 @@
         end
         it "will update appeal state with the 'IhpTaskComplete' status" do
           allow(task).to receive(:verify_user_can_update!).with(user).and_return(true)
+          InitialTasksFactory.new(task.appeal).create_root_and_sub_tasks!
```
Comment (msteele96) [inline spec/models/appellant_notification_spec.rb:949]: I'd like to move this to a before block for the context block. For ease of use you can `let(:appeal) { task.appeal }` in with the rest of them

File: app/models/prepend/va_notify/appellant_notification.rb [hunk 2]
```diff
@@ -23,10 +23,21 @@ def status
...
+    def status
+      "Inactive"
+    end
+  end
+
   class NoAppealError < StandardError; end
 
   def self.handle_errors(appeal)
     fail NoAppealError if appeal.nil?
+    fail InactiveAppealError, appeal["uuid"] if !appeal.active?
```
Comment (msteele96) [inline app/models/prepend/va_notify/appellant_notification.rb:40]: 2 minor changes here:

1. UUID is for AMA only. We need to use external_id which handles both AMA and Legacy
2. When you pass the argument can you use `appeal.external_id` so we're consistent across the codebase

File: app/models/prepend/va_notify/appellant_notification.rb [hunk 3]
```diff
@@ -23,10 +23,21 @@ def status
     end
   end
 
+  class InactiveAppealError < StandardError
+    def initialize(appeal_id, message = "The appeal status is inactive")
+      super(message + " for appeal with id #{appeal_id}")
+    end
+
+    def status
```
Comment (msteele96) [inline app/models/prepend/va_notify/appellant_notification.rb:31]: Does this do anything?

**Review by msteele96 — CHANGES_REQUESTED** [review]

File: app/models/prepend/va_notify/appellant_notification.rb (hunk 3 above)
Comment (noahhansen-gov) [reply app/models/prepend/va_notify/appellant_notification.rb:31]: This helps with the error message `AppellantNotification::InactiveAppealError: The appeal status is inactive for appeal with id 2096907`

**Review by msteele96 — APPROVED** [review]

**Pull request merged**